pkg/commands/workload.go

```diff
@@ -616,21 +618,61 @@ func (opts *WorkloadOptions) Create(ctx context.Context, c *cli.Config, workload
 func (opts *WorkloadOptions) LoadInputWorkload(input io.Reader, workload *cartov1alpha1.Workload) error {
 	var in io.Reader
 
-	f, err := os.Open(opts.FilePath)
-	in = f
-	if f == nil && opts.FilePath == "-" {
+	isURL, err := isUrl(opts.FilePath)
+	if err != nil {
+		return fmt.Errorf("unable to check if filepath %q is a valid url: %w", opts.FilePath, err)
+	}
+	switch {
+	case isURL:
+		in, err = opts.getUrlFileContent()
+		if err != nil {
+			return fmt.Errorf("unable to read from url %q: %w", opts.FilePath, err)
+		}
+	case opts.FilePath == "-":
```

Comment:
**Suggestion**: It is better to modify this to an `if-else` chain:

- as there is an immediate else-if continuation. And
- Case matching is not on the same variable